frontend/pre_analysis.py

```diff
@@ -342,7 +342,14 @@ def add_init_if_not_existing(class_node):
             return
     class_node.body.append(ast.FunctionDef(
```

Comment:
These constructors will be actually, properly added to the program that you output/unparse, right? I'm still not convinced this is the right thing to do, especially since, as long as we don't put super().\_\_init\_\_(*args, **kwargs) in the new \_\_init\_\_, we could actually change the program's behavior with this. 

Isn't it easier to add an \_\_init\_\_ method to class_to_instance_attributes if it's not there instead of adding it to the actual AST?